Sources/SourceKitBazelBSP/SharedUtils/Shell/CommandRunner.swift

```diff
@@ -52,6 +52,18 @@ extension CommandRunner {
         )
     }
 
+    /// A  bazel action query command, but at this BSP's special output base and taking into account the special index flags.
+    func bazelActionQuery(initializedConfig: InitializedServerConfig, cmd: String) throws -> Data {
+        let baseCommand = [
+            initializedConfig.baseConfig.bazelWrapper,
+            "--output_base=\(initializedConfig.outputBase)",
+            cmd,
+        ]
+        let additionalFlags = initializedConfig.baseConfig.indexFlags
+        let command = (baseCommand + additionalFlags).joined(separator: " ")
+        return try run(command, cwd: initializedConfig.rootUri)
+    }
```

Comment:
I haven't understood why this is necessary, is it not the same thing as the function that was being used before?

Reply:
I don't particularly love this method neither, the reason behind is using the existing method requires me to do two overloads. So it is either more overloads or combine them into a standalone method 🤷. Or I can create an extension method called toStr() on Data

Reply:
Ah, it's because of the Data return type, I see now. I _think_ may be able to fix this with by adding a generic return to that method.